lib/bundler/settings/validator.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+module Bundler
+  class Settings
+    class Validator
+      class Rule
+        attr_reader :description
+
+        def initialize(keys, description, &validate)
+          @keys = keys
+          @description = description
+          @validate = validate
+        end
+
+        def validate(key, value, settings)
```

Comment:
`validate` seems to usually do more "dangerous" things like "fail!" and such. Would `validate!` make more sense?